packages/@aws-cdk/aws-ecs-patterns/test/fargate/test.scheduled-fargate-task.ts

```diff
@@ -251,4 +251,33 @@ export = {
 
     test.done();
   },
+
+  "Scheduled Fargate Task - with subnetSelection defined"(test: Test) {
+    // GIVEN
+    const stack = new cdk.Stack();
+    const vpc = new ec2.Vpc(stack, 'Vpc', {
+      maxAzs: 1,
+      natGateways: 0,
+      subnetConfiguration: [
+        { name: 'public', cidrMask: 24, subnetType: ec2.SubnetType.PUBLIC }
+      ],
+    });
+    const cluster = new ecs.Cluster(stack, 'EcsCluster', { vpc });
+
+    new ScheduledFargateTask(stack, 'ScheduledFargateTask', {
+      cluster,
+      scheduledFargateTaskImageOptions: {
+        image: ecs.ContainerImage.fromRegistry('henk'),
+      },
+      subnetSelection: { subnetType: ec2.SubnetType.PUBLIC },
+      schedule: events.Schedule.expression('rate(1 minute)')
+    });
+
+    // THEN
+    expect(stack).to(haveResourceLike('AWS::ECS::TaskDefinition', {
+      ContainerDefinitions: [{ Image: "henk" }]
+    }));
```

Comment:
@floehopper Would it be possible to add an expect statement that checks for the public IP setting?

The other test cases that I see check for a lot more.

Reply:
I've amended the commit to address your feedback on the unit test by changing the assertion to check that the stack has a AWS::Events::Rule configured [to use the public subnet and to have a public IP address][1]. I hope that makes sense.

In doing this I noticed that a number of the other existing tests are a bit confusing in this regard. e.g. the "Scheduled Fargate Task - with MemoryReservation defined" test doesn't actually specify the container memory or assert that it has been set.

Also I think a number of the other tests could be improved by the use of the `haveResourceLike` matcher to reduce the signal-to-noise ratio, e.g. I don't see the point in asserting the presence of the detailed `LogConfiguration` in every test.

If you think these would be good improvements, I'd be happy to tackle them as a separate PR.

[1]: https://github.com/aws/aws-cdk/pull/6624/commits/f8fdc4fca88a59591c396adfb39dd78d7658213d#diff-64a7aca333fa14032084fbd40ac0e30bR282-R286

Reply:
> Also I think a number of the other tests could be improved 

So to be honest I'm very new to the CDK myself, but I agree... I was also a bit confused when I looked at the existing tests.

>  I'd be happy to tackle them as a separate PR.

Go ahead! 🤠